adaptors_task/student/src/impl.cpp

```diff
@@ -1,8 +1,34 @@
 #include "impl.h"
 #include <vector>
+#include <stack>
 
 const std::vector<std::pair<char, char>> brackets {
     {'(', ')'},
     {'[', ']'},
     {'{', '}'}
 };
+
+bool isValid(const std::string& source){
+	std::stack<char> stack;
+	auto find_pair = [](char c){
```

Comment:
1. I'd create a new function for that
2. Try to give it more meaningful name, you're working with brackets, so...